kustomize/commands/internal/kustfile/kustomizationfile.go

```diff
@@ -291,15 +292,16 @@ func findMatchedField(line []byte) (bool, string) {
 // an empty []byte is returned.
 func marshalField(field string, kustomization *types.Kustomization) ([]byte, error) {
 	r := reflect.ValueOf(*kustomization)
-	v := r.FieldByName(strings.Title(field))
+	titleCaser := cases.Title(language.English, cases.NoLower)
+	v := r.FieldByName(titleCaser.String(field))
 
 	if !v.IsValid() || isEmpty(v) {
 		return []byte{}, nil
 	}
 
 	k := &types.Kustomization{}
 	kr := reflect.ValueOf(k)
-	kv := kr.Elem().FieldByName(strings.Title(field))
```

Comment:
This function is deprecated and replaced as implemented. It was straightforward once I realized we needed the `cases.NoLower` option, thanks to the tests.